cou/steps/analyze.py

```diff
@@ -156,12 +159,29 @@ def __str__(self) -> str:
     def min_os_release_apps(apps: list[OpenStackApplication]) -> Optional[OpenStackRelease]:
         """Get the minimal OpenStack release from a list of applications.
 
+        - subordinates or channel based apps are not considered if not using release channels
+        - other apps are considered even if not using release channels
+
         :param apps: Applications.
         :type apps: list[OpenStackApplication]
         :return: OpenStack release.
         :rtype: Optional[OpenStackRelease]
         """
-        return min((app.current_os_release for app in apps), default=None)
+        # NOTE(gabrielcocenza) Apps based on channels to identify OpenStack release cannot
+        # be considered when using latest/stable or charmstore because it's not reliable and
+        # will be considered as Ussuri.
+        apps_skipped = {
+            app
+            for app in apps
+            if isinstance(app, (ChannelBasedApplication, SubordinateBase))
```

Comment:
To me this does not look right to check class here, we should instead bake this to classes instead.
Maybe as class argument?

Reply:
not-blocking: I think it would be better to add something like this to classes.
```python
class SubordianteBase(OpenstackApplication):

    based_on_channel = True
```